src/types/ResetPeriod.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
```

Comment:
Could help to add a comment to ensure this is enum is not getting changed by mistake during development:
// This enum cannot be increased. Else, the first bit will be cut off in the token id

Reply:
So the logic behind using an enum rather than just an arbitrary value (which was the original design) is twofold:
- having the capability to pack all the parameters of the resource lock into the ID is a big win for indexing, and so only taking up 3 bits for the reset period leaves more room for encoding the allocator ID
- standardizing on a set of common reset periods will promote composability so there's a greater chance that a given lock can be transferred directly rather than needing to be withdrawn and redeposited into a new lock